### DISCUSSION TITLE
automatic retries on 500 doesn't happen

### DISCUSSION THREAD

**Discussion by SangeetAgarwal** [vercel/swr#615] [General] [answered]
I have a mutate as follows

`
await mutate(["/api/v1/Resource/UpdateResource"], await axios(config))`

when the server returns a 500, retry doesn't happen.

I'm not sure what I might be missing, any help would be much appreciate

I did specify the following explicitly too
`
        <SWRConfig value={{ refreshInterval: 1000, shouldRetryOnError: true, errorRetryCount: 3}}>
                   <App store={store} userManager={userManager} />
                </SWRConfig>
`

**Comment by sergiodxa** [comment, marked as answer]
The retry feature is only for the useSWR hook, it's not available within the `mutate` function, if you use:
```
useSWR(key, fetcher)
```
And your fetcher throw an error It will retry it.

If you use
```
mutate(key, fetcher)
```
And your fetcher throw an error it will not retry and it will update the cache with the error.

You can implement retry yourself inside your fetcher function, an easy way is with [async-retry](https://github.com/vercel/async-retry).
